WhenWorkingWithAPetQueue.java

```diff
@@ -0,0 +1,42 @@
+package serenitylabs.tutorials.vetclinic.collections.exercises;
+
+import org.junit.Assert;
+import org.junit.Test;
+import serenitylabs.tutorials.vetclinic.Pet;
+
+import java.util.LinkedList;
+import java.util.Queue;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import static org.hamcrest.core.IsCollectionContaining.hasItem;
+import static org.hamcrest.core.IsEqual.equalTo;
+
+
+public class WhenWorkingWithAPetQueue {
+
+    @Test
+    public void should_add_Fido_to_the_end_of_the_queue() {
+        Queue<Pet> waitingList = new LinkedList<>();
+
+        waitingList.add(Pet.cat().named("Felix"));
+        waitingList.add(Pet.dog().named("Fido"));
+
+        LinkedList<Pet> listOfPets = (LinkedList) waitingList;
+
+        assertThat("Fido", equalTo(listOfPets.getLast().getName()));
```

Comment:
We can also use the poll() method to ensure that Felix is still at the front of the line:
```
Pet nextInLine = waitingList.poll();
assertThat(nextInLine.getName(), equalTo("Felix"));
```